task/ai-code-review/0.1/ai-code-review.yaml

```diff
@@ -0,0 +1,484 @@
+                logger.info(f"Processing PR #{pr_details['pull_number']} in {pr_details['owner']}/{pr_details['repo']}")
+
+                # Get diff
+                diff = get_diff(pr_details["owner"], pr_details["repo"], pr_details["pull_number"], args.token, args.github_host)
+
+                if not diff:
+                    logger.warning("No diff found.")
+                    return
+
+                parsed_diff = parse_diff(diff)
+
+                # Apply exclusion patterns
+                exclude_patterns = [p.strip() for p in args.exclude.split(",") if p.strip()]
+                filtered_diff = [
+                    file for file in parsed_diff
+                    if not any(fnmatch_pattern(file.get("to", ""), pattern) for pattern in exclude_patterns)
+                ]
+
+                logger.info(f"Analyzing {len(filtered_diff)} files after applying exclusions.")
+
+                # Analyze code
+                comments = analyze_code(filtered_diff, pr_details, args.api_base_url, args.api_key, args.api_model)
+
+                logger.info(f"Generated {len(comments)} comment(s).")
+
+                for comment in comments:
+                    logger.debug(f"File: {comment['path']}, Line: {comment['line']}\nComment: {comment['body']}\n")
+
+                if not args.dry_run and comments:
+                    create_review_comment(
+                        pr_details["owner"],
+                        pr_details["repo"],
+                        pr_details["pull_number"],
+                        comments,
+                        args.token,
+                        args.github_host
+                    )
+                    logger.info("Review comments posted successfully.")
+                elif args.dry_run:
+                    logger.info("Dry run enabled — comments not posted.")
+                    for comment in comments:
+                        print(f"File: {comment['path']}, Line: {comment['line']}")
+                        print(f"Comment: {comment['body']}\n")
+                else:
+                    logger.info("No issues found to comment on.")
+
+            except Exception as e:
+                logger.exception("An error occurred during execution:")
+                import sys
+                sys.exit(1)
+
+        if __name__ == "__main__":
+            main()
+
+        EOF
+        
+        chmod +x code-review.py
+
+    - name: install-dependencies
+      image: docker.io/library/python:3.9-slim
+      workingDir: $(workspaces.shared-workspace.path)
+      script: |
+        python -m venv venv
+        # shellcheck source=venv/bin/activate disable=SC1091
+        . venv/bin/activate
+        pip install --upgrade pip
+        pip install requests openai httpx PyGithub
+    
+    - name: run-code-review
+      image: docker.io/library/python:3.9-slim
+      workingDir: $(workspaces.shared-workspace.path)
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
```

Comment:
should we consider merging these two steps as two steps would be equal to two containers? This task already has 3 steps which makes it 3 containers 😅

Reply:
fixed